crates/astria-sequencer/src/grpc/sequencer.rs

```diff
@@ -152,6 +168,102 @@ impl SequencerService for SequencerServer {
 
         Ok(Response::new(block))
     }
+
+    #[instrument(skip_all)]
+    async fn get_chain_id(
+        self: Arc<Self>,
+        _: Request<GetChainIdRequest>,
+    ) -> Result<Response<ChainId>, Status> {
+        let snapshot = self.storage.latest_snapshot();
+
+        let chain_id = snapshot
+            .get_chain_id()
+            .await
+            .map_err(|e| Status::internal(format!("failed to get chain id from storage: {e}")))?;
+
+        Ok(Response::new(ChainId {
+            inner: chain_id.to_string(),
+        }))
+    }
+
+    #[instrument(skip_all)]
+    async fn get_commit(
+        self: Arc<Self>,
+        request: Request<GetCommitRequest>,
+    ) -> Result<Response<Commit>, Status> {
+        let request = request.into_inner();
+        self.validate_height(request.height).await?;
+
+        let height: u32 = request
+            .height
+            .try_into()
+            .map_err(|_| Status::invalid_argument("height is too large to fit into an u32"))?;
+
+        let resp =
+            self.cometbft_client.commit(height).await.map_err(|e| {
+                Status::internal(format!("failed to get commit from cometbft: {e}"))
+            })?;
+
+        let resp = Commit {
+            signed_header: Some(resp.signed_header.into()),
+            canonical: resp.canonical,
+        };
+
+        Ok(Response::new(resp))
+    }
+
+    #[instrument(skip_all)]
+    async fn get_validators(
+        self: Arc<Self>,
+        request: Request<GetValidatorsRequest>,
+    ) -> Result<Response<Validators>, Status> {
+        let request = request.into_inner();
+        self.validate_height(request.height).await?;
+
+        let height: u32 = request
+            .height
+            .try_into()
+            .map_err(|_| Status::invalid_argument("height is too large to fit into an u32"))?;
+
+        let resp = self
+            .cometbft_client
+            .validators(height, Paging::Default)
```

Comment:
Should we be requesting `Paging::All` here, or maybe we need to implement paging on our server too?